### DISCUSSION TITLE
Replace all colours to specific colour

### DISCUSSION THREAD

**Discussion by defunky** [dlemstra/Magick.NET#1175] [Help] [answered]
I have a regular image with multiple colours, what is the most efficient way to replace all these colours to a single colour, e.g. all pixels non-alpha pixels become red.

I found Opaque() method but that targets a single colour only.

**Comment by dlemstra** [comment, marked as answer]
One option could be to use the `Fx` method:

```C#
using var image = new MagickImage("input.png");
image.Fx("a == 0 ? a : red");
image.Write("output.png");
```

Some documentation about that can be found here: https://imagemagick.org/script/fx.php.
